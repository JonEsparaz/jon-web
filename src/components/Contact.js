import React from 'react';
import './Contact.scss';

function Contact() {
  return (
    <div className="ContactContainer">
    <a href = "mailto:jon.esparaz@gmail.com">Email</a>
    </div>
  );
}

export default Contact;
