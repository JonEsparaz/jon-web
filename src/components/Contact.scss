    .ContactContainer {
        padding-top: 10vw;
        text-align: center;
        position: relative;
        min-height: 100vh;
        font-family: Ubuntu;
        background: #1a1a1a;
        color: white;
    }

    .ContactText {
        color: white;
    }