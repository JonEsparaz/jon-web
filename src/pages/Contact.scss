@import '../theme.scss';

@keyframes SentIcon {
  0% {
      width: 0px;
      height: 0px;
      transform: rotate(-270deg);
  }
  100% {
      transform: rotate(0deg);
  }
}

.Contact {
  animation: FadeIn 1s;
}

@media (min-width: 768px) {
  .CheckIcon {
    animation: SentIcon 0.9s;
    width: 48px;
    height: 48px;
  }

  .ContactContainer {
    text-align: center;
    position: relative;
    font-family: $font;
    margin-left: 27.5vw;
    margin-top: 4vw;
    min-height: calc(100vh - 106px - 159px - 4vw);
  }

  .ContactForm {
    text-align: left;
  }

  .NameInputContainer {
    display: flex;
    width: 45vw;
    flex-direction: row;
  }

  .NameInput {
    flex: 1;
    height: 2vw;
    font-family: $font;
    font-size: 1vw;
    margin-bottom: 1vw;
    border: 1px black solid;
    background: #f5f5f5;
  }

  .FormInput {
    width: 45vw;
    height: 2vw;
    font-family: $font;
    font-size: 1vw;
    margin-bottom: 1vw;
    border: 1px black solid;
    background: #f5f5f5;
  }

  .MessageInput {
    width: 45vw;
    height: 4vw;
    font-family: $font;
    font-size: 1vw;
    border: 1px black solid;
    resize: vertical;
    margin-top: 1vw;
    background: #f5f5f5;
  }
}

@media (max-width: 767px) {
  .CheckIcon {
    animation: SentIcon 0.9s;
    width: 10vw;
    height: 10vw;
  }

  .ContactContainer {
    text-align: center;
    position: relative;
    font-family: $font;
    margin-top: 4vw;
    min-height: calc(100vh - 106px - 159px - 4vw);
  }

  .ContactForm {
    text-align: left;
    width: 90vw;
    margin-left: 5vw;
    margin-right: 5vw;
  }

  .NameInputContainer {
    display: flex;
    flex-direction: row;
  }

  .NameInput {
    flex: 1;
    height: 10vw;
    width: 40vw;
    font-family: $font;
    font-size: 5vw;
    margin-bottom: 3vw;
    border: 1px black solid;
    background: #f5f5f5;
  }

  .FormInput {
    width: 90vw;
    height: 10vw;
    font-family: $font;
    font-size: 5vw;
    margin-bottom: 3vw;
    border: 1px black solid;
    background: #f5f5f5;
  }

  .MessageInput {
    width: 90vw;
    height: 20vw;
    font-family: $font;
    font-size: 5vw;
    border: 1px black solid;
    resize: vertical;
    margin-top: 3vw;
    background: #f5f5f5;
  }
}