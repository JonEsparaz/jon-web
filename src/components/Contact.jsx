import React from 'react';
import './Contact.scss';

function Contact() {
  return (
    <div className="ContactContainer">
    <div>I'm still building a contact form. Feel free to email me directly: &nbsp;<a className="ContactText" href = "mailto:jon.esparaz@gmail.com"> jon.esparaz@gmail.com</a></div>
    </div>
  );
}

export default Contact;
