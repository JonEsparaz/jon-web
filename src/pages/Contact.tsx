import React, { useState } from 'react';
import * as Sentry from '@sentry/react';
import { API } from 'aws-amplify';
import { GRAPHQL_AUTH_MODE } from '@aws-amplify/api';
import ReCAPTCHA from 'react-google-recaptcha';
import Menu from '../components/Menu';
import Footer from '../components/Footer';
import { ContactQueryVariables } from '../API';
import { contact } from '../graphql/queries';
import './Contact.scss';

export default function Contact(): JSX.Element {
  const [isSent, setIsSent] = useState(false);
  const [error, setError] = useState(false);
  const [emailObj, setEmailObj] = useState<ContactQueryVariables>({
    email: '',
    first: '',
    last: '',
    subject: '',
    message: '',
  });

  const isLocalhost = window.location.hostname === 'localhost';
  const recaptchaRef = React.createRef<ReCAPTCHA>();

  const send = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();

    const recaptchaValue = recaptchaRef.current?.getValue();

    if (recaptchaValue || isLocalhost) {
      try {
        await API.graphql({
          query: contact,
          variables: emailObj,
          authMode: GRAPHQL_AUTH_MODE.API_KEY,
        });

        setEmailObj({
          email: '',
          first: '',
          last: '',
          subject: '',
          message: '',
        });
        setIsSent(true);
      } catch (err) {
        Sentry.captureException(err);
        setError(true);
      }
    }
  };

  const handleChange = (field: keyof ContactQueryVariables, input: string) => {
    setEmailObj((prevState) => ({
      ...prevState,
      [field]: input,
    }));
  };

  return (
    <div className="page-wrapper">
      <Menu mode="dark" />
      <div className="ContactContainer center-wrapper page-body">
        <main>
          <form className="ContactForm" onSubmit={(e) => send(e)}>
            <div className="NameInputContainer">
              <input
                aria-label="First Name"
                style={{ marginRight: 10 }}
                className="NameInput"
                type="text"
                placeholder="First name"
                required
                value={emailObj.first}
                onChange={(e) => handleChange('first', e.target.value)}
              />
              <input
                aria-label="Last Name"
                className="NameInput"
                type="text"
                placeholder="Last name"
                required
                value={emailObj.last}
                onChange={(e) => handleChange('last', e.target.value)}
              />
            </div>
            <input
              aria-label="Email Address"
              className="FormInput"
              type="email"
              placeholder="Email"
              required
              value={emailObj.email}
              onChange={(e) => handleChange('email', e.target.value)}
            />
            <br />
            <input
              aria-label="Subject Line"
              className="FormInput"
              type="text"
              placeholder="Subject"
              required
              value={emailObj.subject}
              onChange={(e) => handleChange('subject', e.target.value)}
            />
            <br />
            <textarea
              aria-label="Message"
              className="MessageInput"
              placeholder="Message..."
              required
              value={emailObj.message}
              onChange={(e) => handleChange('message', e.target.value)}
            />
            <br />
            <ReCAPTCHA
              sitekey={
                isLocalhost
                  ? '6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI'
                  : '6Lcq77QZAAAAAPuPm5u1lqeo-p_EyXxxLymBS-ZK'
              }
              ref={recaptchaRef}
            />
            <button
              className="ActionButton2"
              type="submit"
              disabled={isSent}
              data-testid="send-btn"
            >
              <span className={isSent ? '' : 'Underline2'}>
                {isSent ? 'Sent' : 'Send'}
              </span>
            </button>
            {isSent && (
              <span style={{ marginLeft: 12 }}>
                <img
                  data-testid="success-icon"
                  src="/svg/check_circle_outline.svg"
                  alt="message sent"
                  className="CheckIcon"
                />
              </span>
            )}
            {error && (
              <span style={{ marginLeft: 12 }}>
                <img
                  data-testid="error-icon"
                  src="/svg/error_outline.svg"
                  alt="message failed to send"
                  className="CheckIcon"
                />
              </span>
            )}
          </form>
        </main>
      </div>
      <Footer />
    </div>
  );
}
