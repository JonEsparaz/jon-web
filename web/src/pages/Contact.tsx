import React, { useState } from 'react';
import * as Sentry from '@sentry/react';
import { Form, FormGroup, Label, Input } from 'reactstrap';
import { API } from 'aws-amplify';
import ReCAPTCHA from 'react-google-recaptcha';
import Page from '../components/Page';
import Button from '../components/Button';
import { ContactQueryVariables } from '../API';
import { contact } from '../graphql/queries';
import './Contact.scss';

export default function Contact(): JSX.Element {
  const isLocalhost = window.location.hostname === 'localhost';

  const [isSent, setIsSent] = useState(false);
  const [error, setError] = useState(false);
  const [emailObj, setEmailObj] = useState<ContactQueryVariables>({
    email: '',
    first: '',
    last: '',
    subject: '',
    message: '',
  });

  const recaptchaRef = React.createRef<ReCAPTCHA>();

  const send = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();

    const recaptchaValue = recaptchaRef.current?.getValue();

    if (recaptchaValue || isLocalhost) {
      try {
        await API.graphql({
          query: contact,
          variables: emailObj,
        });

        setEmailObj({
          email: '',
          first: '',
          last: '',
          subject: '',
          message: '',
        });
        setIsSent(true);
      } catch (err) {
        Sentry.captureException(err);
        setError(true);
      }
    }
  };

  const handleChange = (field: keyof ContactQueryVariables, input: string) => {
    setEmailObj((prevState) => ({
      ...prevState,
      [field]: input,
    }));
  };

  return (
    <Page mode="dark">
      <div className="container">
        <div className="mx-1 mb-4">
          <main>
            <Form className="col-sm-6 m-auto" onSubmit={(e) => send(e)}>
              <FormGroup className="row mb-2">
                <div className="col-6">
                  <Label for="fname">First name</Label>
                  <Input
                    required
                    value={emailObj.first}
                    onChange={(e) => handleChange('first', e.target.value)}
                    id="fname"
                    className="rounded-0"
                  />
                </div>
                <div className="col-6">
                  <Label for="lname">Last name</Label>
                  <Input
                    required
                    value={emailObj.last}
                    onChange={(e) => handleChange('last', e.target.value)}
                    id="lname"
                    className="rounded-0"
                  />
                </div>
              </FormGroup>
              <FormGroup className="mb-2">
                <Label for="email">Email</Label>
                <Input
                  required
                  type="email"
                  value={emailObj.email}
                  onChange={(e) => handleChange('email', e.target.value)}
                  id="email"
                  className="rounded-0"
                />
              </FormGroup>
              <FormGroup className="mb-2">
                <Label for="subject">Subject</Label>
                <Input
                  required
                  value={emailObj.subject}
                  onChange={(e) => handleChange('subject', e.target.value)}
                  id="subject"
                  className="rounded-0"
                />
              </FormGroup>
              <FormGroup className="mb-4">
                <Label for="message">Message</Label>
                <Input
                  required
                  type="textarea"
                  value={emailObj.message}
                  onChange={(e) => handleChange('message', e.target.value)}
                  id="message"
                  className="rounded-0"
                />
              </FormGroup>
              <ReCAPTCHA
                sitekey="6Lcq77QZAAAAAPuPm5u1lqeo-p_EyXxxLymBS-ZK"
                ref={recaptchaRef}
              />
              <Button dark type="submit" disabled={isSent} testId="send-btn">
                {isSent ? 'Sent' : 'Send'}
              </Button>
              {isSent && (
                <span className="ml-3">
                  <img
                    data-testid="success-icon"
                    src="/svg/check_circle_outline.svg"
                    alt="check mark: message sent"
                    className="email-status-icon"
                  />
                </span>
              )}
              {error && (
                <span className="ml-3">
                  <img
                    data-testid="error-icon"
                    src="/svg/error_outline.svg"
                    alt="error: message failed to send"
                    className="email-status-icon"
                  />
                </span>
              )}
            </Form>
          </main>
        </div>
      </div>
    </Page>
  );
}
