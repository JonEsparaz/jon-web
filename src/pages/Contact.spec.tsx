import React from 'react';
import { API } from '@aws-amplify/api';
import { fireEvent, render, act } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Contact from './Contact';

beforeAll(() => {
  jest.clearAllMocks();
});

describe('contact form', () => {
  test('snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <Contact />
      </MemoryRouter>,
    );

    expect(container.innerHTML).toMatchSnapshot();
  });

  test('api error', async () => {
    const { getByTestId, getByLabelText, container } = render(
      <MemoryRouter>
        <Contact />
      </MemoryRouter>,
    );

    const firstName = getByLabelText('First name') as HTMLInputElement;
    fireEvent.change(firstName, { target: { value: 'Luke' } });
    expect(firstName.value).toBe('Luke');

    const lastName = getByLabelText('Last name') as HTMLInputElement;
    fireEvent.change(lastName, { target: { value: 'Skywalker' } });
    expect(lastName.value).toBe('Skywalker');

    const emailAddr = getByLabelText('Email') as HTMLInputElement;
    fireEvent.change(emailAddr, {
      target: { value: 'luke.skywalker@jedi.com' },
    });
    expect(emailAddr.value).toBe('luke.skywalker@jedi.com');

    const subject = getByLabelText('Subject') as HTMLInputElement;
    fireEvent.change(subject, { target: { value: 'This is a test' } });
    expect(subject.value).toBe('This is a test');

    const message = getByLabelText('Message') as HTMLInputElement;
    fireEvent.change(message, { target: { value: 'Test message' } });
    expect(message.value).toBe('Test message');

    API.graphql = jest.fn().mockRejectedValueOnce(() => new Error('error'));

    await act(async () => {
      fireEvent.click(getByTestId('send-btn'));
    });

    expect(getByTestId('error-icon')).toBeTruthy();
    expect(container.innerHTML).toMatchSnapshot();
  });

  test('submit message successfully', async () => {
    const { getByTestId, getByLabelText, container } = render(
      <MemoryRouter>
        <Contact />
      </MemoryRouter>,
    );

    const firstName = getByLabelText('First name') as HTMLInputElement;
    fireEvent.change(firstName, { target: { value: 'Luke' } });
    expect(firstName.value).toBe('Luke');

    const lastName = getByLabelText('Last name') as HTMLInputElement;
    fireEvent.change(lastName, { target: { value: 'Skywalker' } });
    expect(lastName.value).toBe('Skywalker');

    const emailAddr = getByLabelText('Email') as HTMLInputElement;
    fireEvent.change(emailAddr, {
      target: { value: 'luke.skywalker@jedi.com' },
    });
    expect(emailAddr.value).toBe('luke.skywalker@jedi.com');

    const subject = getByLabelText('Subject') as HTMLInputElement;
    fireEvent.change(subject, { target: { value: 'This is a test' } });
    expect(subject.value).toBe('This is a test');

    const message = getByLabelText('Message') as HTMLInputElement;
    fireEvent.change(message, { target: { value: 'Test message' } });
    expect(message.value).toBe('Test message');

    API.graphql = jest.fn().mockResolvedValueOnce(() => {});

    // form complete
    await act(async () => {
      fireEvent.click(getByTestId('send-btn'));
    });

    expect(API.graphql).toHaveBeenCalled();
    expect(getByTestId('success-icon')).toBeTruthy();
    expect(container.innerHTML).toMatchSnapshot();
  });
});
