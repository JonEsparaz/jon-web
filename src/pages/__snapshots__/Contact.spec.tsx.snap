// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact form api error 1`] = `"<div class=\\"page-wrapper\\"><div class=\\"MenuContainer\\" style=\\"position: relative;\\"><nav class=\\"navbar navbar-expand-md\\"><a href=\\"/\\"><img alt=\\"logo\\" src=\\"/logos/je-logo.png\\" style=\\"height: 80px; margin-right: 2vw;\\"></a><button aria-label=\\"Toggle navigation\\" type=\\"button\\" class=\\"navbar-toggler\\"><div class=\\"hamburger hamburger--slider  bl-menu\\"><div class=\\"hamburger-box\\"><div class=\\"hamburger-inner\\"></div></div></div></button><div class=\\"collapse navbar-collapse\\"><ul class=\\"mr-auto navbar-nav\\"><li><a class=\\"navlink-custom\\" href=\\"/speedcubing\\"><img src=\\"/svg/cubing-black.svg\\" alt=\\"\\" class=\\"menu-icon\\"><span class=\\"Underline\\">Speedcubing</span></a></li><li><a class=\\"navlink-custom\\" href=\\"/contact\\"><img src=\\"/svg/email-black.svg\\" alt=\\"\\" class=\\"menu-icon\\"><span class=\\"Underline\\">Contact</span></a></li></ul></div></nav></div><div class=\\"ContactContainer center-wrapper page-body\\"><main><form class=\\"ContactForm\\"><div class=\\"NameInputContainer\\"><input aria-label=\\"First Name\\" style=\\"margin-right: 10px;\\" class=\\"NameInput\\" type=\\"text\\" placeholder=\\"First name\\" required=\\"\\" value=\\"Luke\\"><input aria-label=\\"Last Name\\" class=\\"NameInput\\" type=\\"text\\" placeholder=\\"Last name\\" required=\\"\\" value=\\"Skywalker\\"></div><input aria-label=\\"Email Address\\" class=\\"FormInput\\" type=\\"email\\" placeholder=\\"Email\\" required=\\"\\" value=\\"luke.skywalker@jedi.com\\"><br><input aria-label=\\"Subject Line\\" class=\\"FormInput\\" type=\\"text\\" placeholder=\\"Subject\\" required=\\"\\" value=\\"This is a test\\"><br><textarea aria-label=\\"Message\\" class=\\"MessageInput\\" placeholder=\\"Message...\\" required=\\"\\">Test message</textarea><br><div></div><button class=\\"ActionButton2\\" type=\\"submit\\" data-testid=\\"send-btn\\"><span class=\\"Underline2\\">Send</span></button><span style=\\"margin-left: 12px;\\"><img data-testid=\\"error-icon\\" src=\\"/svg/error_outline.svg\\" alt=\\"message failed to send\\" class=\\"CheckIcon\\"></span></form></main></div><div class=\\"FooterContainer\\"><a aria-label=\\"Twitter\\" href=\\"https://twitter.com/JonEsparaz\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/twitter.svg\\" alt=\\"\\" class=\\"footer-icon\\"></a><a aria-label=\\"GitHub\\" href=\\"https://github.com/JonEsparaz\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/github.svg\\" alt=\\"\\" class=\\"footer-icon\\"></a><a aria-label=\\"LinkedIn\\" href=\\"https://www.linkedin.com/in/jonathanesparaz/\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/linkedin.svg\\" alt=\\"\\" class=\\"footer-icon\\"></a><a aria-label=\\"YouTube\\" href=\\"https://www.youtube.com/c/JonEsparaz\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/youtube.svg\\" alt=\\"\\" class=\\"footer-icon\\"></a><div style=\\"margin-top: 20px; color: black;\\">© 2021 Jonathan Esparaz</div></div></div>"`;

exports[`contact form snapshot 1`] = `"<div class=\\"page-wrapper\\"><div class=\\"MenuContainer\\" style=\\"position: relative;\\"><nav class=\\"navbar navbar-expand-md\\"><a href=\\"/\\"><img alt=\\"logo\\" src=\\"/logos/je-logo.png\\" style=\\"height: 80px; margin-right: 2vw;\\"></a><button aria-label=\\"Toggle navigation\\" type=\\"button\\" class=\\"navbar-toggler\\"><div class=\\"hamburger hamburger--slider  bl-menu\\"><div class=\\"hamburger-box\\"><div class=\\"hamburger-inner\\"></div></div></div></button><div class=\\"collapse navbar-collapse\\"><ul class=\\"mr-auto navbar-nav\\"><li><a class=\\"navlink-custom\\" href=\\"/speedcubing\\"><img src=\\"/svg/cubing-black.svg\\" alt=\\"\\" class=\\"menu-icon\\"><span class=\\"Underline\\">Speedcubing</span></a></li><li><a class=\\"navlink-custom\\" href=\\"/contact\\"><img src=\\"/svg/email-black.svg\\" alt=\\"\\" class=\\"menu-icon\\"><span class=\\"Underline\\">Contact</span></a></li></ul></div></nav></div><div class=\\"ContactContainer center-wrapper page-body\\"><main><form class=\\"ContactForm\\"><div class=\\"NameInputContainer\\"><input aria-label=\\"First Name\\" style=\\"margin-right: 10px;\\" class=\\"NameInput\\" type=\\"text\\" placeholder=\\"First name\\" required=\\"\\" value=\\"\\"><input aria-label=\\"Last Name\\" class=\\"NameInput\\" type=\\"text\\" placeholder=\\"Last name\\" required=\\"\\" value=\\"\\"></div><input aria-label=\\"Email Address\\" class=\\"FormInput\\" type=\\"email\\" placeholder=\\"Email\\" required=\\"\\" value=\\"\\"><br><input aria-label=\\"Subject Line\\" class=\\"FormInput\\" type=\\"text\\" placeholder=\\"Subject\\" required=\\"\\" value=\\"\\"><br><textarea aria-label=\\"Message\\" class=\\"MessageInput\\" placeholder=\\"Message...\\" required=\\"\\"></textarea><br><div></div><button class=\\"ActionButton2\\" type=\\"submit\\" data-testid=\\"send-btn\\"><span class=\\"Underline2\\">Send</span></button></form></main></div><div class=\\"FooterContainer\\"><a aria-label=\\"Twitter\\" href=\\"https://twitter.com/JonEsparaz\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/twitter.svg\\" alt=\\"\\" class=\\"footer-icon\\"></a><a aria-label=\\"GitHub\\" href=\\"https://github.com/JonEsparaz\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/github.svg\\" alt=\\"\\" class=\\"footer-icon\\"></a><a aria-label=\\"LinkedIn\\" href=\\"https://www.linkedin.com/in/jonathanesparaz/\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/linkedin.svg\\" alt=\\"\\" class=\\"footer-icon\\"></a><a aria-label=\\"YouTube\\" href=\\"https://www.youtube.com/c/JonEsparaz\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/youtube.svg\\" alt=\\"\\" class=\\"footer-icon\\"></a><div style=\\"margin-top: 20px; color: black;\\">© 2021 Jonathan Esparaz</div></div></div>"`;

exports[`contact form submit message successfully 1`] = `"<div class=\\"page-wrapper\\"><div class=\\"MenuContainer\\" style=\\"position: relative;\\"><nav class=\\"navbar navbar-expand-md\\"><a href=\\"/\\"><img alt=\\"logo\\" src=\\"/logos/je-logo.png\\" style=\\"height: 80px; margin-right: 2vw;\\"></a><button aria-label=\\"Toggle navigation\\" type=\\"button\\" class=\\"navbar-toggler\\"><div class=\\"hamburger hamburger--slider  bl-menu\\"><div class=\\"hamburger-box\\"><div class=\\"hamburger-inner\\"></div></div></div></button><div class=\\"collapse navbar-collapse\\"><ul class=\\"mr-auto navbar-nav\\"><li><a class=\\"navlink-custom\\" href=\\"/speedcubing\\"><img src=\\"/svg/cubing-black.svg\\" alt=\\"\\" class=\\"menu-icon\\"><span class=\\"Underline\\">Speedcubing</span></a></li><li><a class=\\"navlink-custom\\" href=\\"/contact\\"><img src=\\"/svg/email-black.svg\\" alt=\\"\\" class=\\"menu-icon\\"><span class=\\"Underline\\">Contact</span></a></li></ul></div></nav></div><div class=\\"ContactContainer center-wrapper page-body\\"><main><form class=\\"ContactForm\\"><div class=\\"NameInputContainer\\"><input aria-label=\\"First Name\\" style=\\"margin-right: 10px;\\" class=\\"NameInput\\" type=\\"text\\" placeholder=\\"First name\\" required=\\"\\" value=\\"\\"><input aria-label=\\"Last Name\\" class=\\"NameInput\\" type=\\"text\\" placeholder=\\"Last name\\" required=\\"\\" value=\\"\\"></div><input aria-label=\\"Email Address\\" class=\\"FormInput\\" type=\\"email\\" placeholder=\\"Email\\" required=\\"\\" value=\\"\\"><br><input aria-label=\\"Subject Line\\" class=\\"FormInput\\" type=\\"text\\" placeholder=\\"Subject\\" required=\\"\\" value=\\"\\"><br><textarea aria-label=\\"Message\\" class=\\"MessageInput\\" placeholder=\\"Message...\\" required=\\"\\"></textarea><br><div></div><button class=\\"ActionButton2\\" type=\\"submit\\" data-testid=\\"send-btn\\" disabled=\\"\\"><span class=\\"\\">Sent</span></button><span style=\\"margin-left: 12px;\\"><img data-testid=\\"success-icon\\" src=\\"/svg/check_circle_outline.svg\\" alt=\\"message sent\\" class=\\"CheckIcon\\"></span></form></main></div><div class=\\"FooterContainer\\"><a aria-label=\\"Twitter\\" href=\\"https://twitter.com/JonEsparaz\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/twitter.svg\\" alt=\\"\\" class=\\"footer-icon\\"></a><a aria-label=\\"GitHub\\" href=\\"https://github.com/JonEsparaz\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/github.svg\\" alt=\\"\\" class=\\"footer-icon\\"></a><a aria-label=\\"LinkedIn\\" href=\\"https://www.linkedin.com/in/jonathanesparaz/\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/linkedin.svg\\" alt=\\"\\" class=\\"footer-icon\\"></a><a aria-label=\\"YouTube\\" href=\\"https://www.youtube.com/c/JonEsparaz\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/youtube.svg\\" alt=\\"\\" class=\\"footer-icon\\"></a><div style=\\"margin-top: 20px; color: black;\\">© 2021 Jonathan Esparaz</div></div></div>"`;
